File: pdu/python/sensor_msgs/pdu_cdr_conv_JoyFeedback.py
from .pdu_pytype_JoyFeedback import JoyFeedback
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_JoyFeedback(writer: CdrWriter, src: JoyFeedback):
    writer.write_uint8(src.type)
    writer.write_uint8(src.id)
    writer.write_float32(src.intensity)


def cdr_body_to_py_JoyFeedback(reader: CdrReader, dst: JoyFeedback):
    dst.type = reader.read_uint8()
    dst.id = reader.read_uint8()
    dst.intensity = reader.read_float32()
    return dst


def py_to_cdr_JoyFeedback(src: JoyFeedback) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_JoyFeedback(writer, src)
    return writer.bytes()


def cdr_to_py_JoyFeedback(cdr_payload) -> JoyFeedback:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = JoyFeedback()
    return cdr_body_to_py_JoyFeedback(reader, dst)

File: pdu/javascript/sensor_msgs/pdu_cdr_conv_JoyFeedback.js
import { JoyFeedback } from './pdu_jstype_JoyFeedback.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduJoyFeedbackConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { JoyFeedback } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint8(src.type);
        writer.write_uint8(src.id);
        writer.write_float32(src.intensity);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { JoyFeedback } dst
     * @returns { JoyFeedback }
     */
    static cdr_body_to_js(reader, dst) {
        dst.type = reader.read_uint8();
        dst.id = reader.read_uint8();
        dst.intensity = reader.read_float32();
        return dst;
    }

    /**
     * @param { JoyFeedback } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { JoyFeedback }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new JoyFeedback());
    }
}
